atc/creds/vault/vault_kvhelpers.go

```diff
@@ -0,0 +1,100 @@
+package vault
+
+import (
+	"errors"
+	"path"
+	"strings"
+
+	"github.com/hashicorp/vault/api"
+)
+
+// The below helper functions are taken from the github.com/hashicorp/vault repository
+// as they are not exposed natively.
+// https://github.com/hashicorp/vault/blob/4b790d2c42f406a980230c196eeeb5b9b52a7cf1/command/kv_helpers.go#L44-L116
+
+func kvPreflightVersionRequest(client *api.Client, path string) (string, int, error) {
+	// We don't want to use a wrapping call here so save any custom value and
+	// restore after
+	currentWrappingLookupFunc := client.CurrentWrappingLookupFunc()
+	client.SetWrappingLookupFunc(nil)
+	defer client.SetWrappingLookupFunc(currentWrappingLookupFunc)
+	currentOutputCurlString := client.OutputCurlString()
+	client.SetOutputCurlString(false)
+	defer client.SetOutputCurlString(currentOutputCurlString)
```

Comment:
Good call - those functions are mainly used for vault cli (OutputCurlString) and the [Cubbyhole Wrapping](https://www.vaultproject.io/docs/concepts/response-wrapping) functions of vault, neither of which concourse would need to use to implement kv_v2 support. Initially I wanted to avoid modifying upstream functions in case they need to be updated in the future, however this makes sense and does not impact functionality from the testing i've done.